src/core/index.js

```diff
@@ -43,21 +46,24 @@ function IPFS (repoInstance) {
   this._blockS = new BlockService(this._repo)
   this._dagS = new DAGService(this._blockS)
 
+  // IPFS Core exposed modules
+  //   for booting up a node
   this.goOnline = goOnline(this)
   this.goOffline = goOffline(this)
   this.isOnline = isOnline(this)
   this.load = load(this)
+  this.init = init(this)
+  //   interface-ipfs-core defined API
   this.version = version(this)
   this.id = id(this)
   this.repo = repo(this)
-  this.init = init(this)
   this.bootstrap = bootstrap(this)
   this.config = config(this)
   this.block = block(this)
   this.object = object(this)
   this.libp2p = libp2p(this)
-  this.swarm = this.libp2p.swarm // for interface-ipfs-core sake
+  this.swarm = swarm(this)
   this.files = files(this)
-  this.cat = files(this).cat // Alias for js-ipfs-api cat
   this.bitswap = bitswap(this)
+  this.ping = ping(this)
```

Comment:
This should make it more obvious for folks to understand why this pieces are all attached here.